projects/core/directives/hint/hint.component.ts

```diff
@@ -104,7 +104,8 @@ export class TuiHintComponent<C = any> {
 
     protected onClick(target: HTMLElement): void {
         if (
-            (!target.closest('tui-hint') && !this.hint.el.contains(target)) ||
+            (!target.closest(this.el.tagName.toLowerCase()) &&
```

Comment:
It's case insensitive, no need for lowercasing

Reply:
fixed